services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/team/TeamManagerImplTest.java

```diff
@@ -773,21 +773,40 @@ public void testGetIconURL() {
 	}
 
 	@Test
-	public void testGetIconURLUserInfo() {
+	public void testGetIconURLWithNullUserInfo() {
 		Exception exception = assertThrows(IllegalArgumentException.class, () -> {
 			teamManagerImpl.getIconURL(null, TEAM_ID);
 		});
 		assertEquals("userInfo is required.", exception.getMessage());
 	}
 
 	@Test
-	public void testGetIconURLTeamId() {
+	public void testGetIconURLWithNullTeamId() {
 		Exception exception = assertThrows(IllegalArgumentException.class, () -> {
 			teamManagerImpl.getIconURL(userInfo, null);
 		});
 		assertEquals("teamId is required.", exception.getMessage());
 	}
 
+	@Test
+	public void testGetIconPreviewURLPreviewFileHandleIdNotFoundException() {
+
+		NotFoundException cause = new NotFoundException("inner exception");
+		String iconFileHandleId = "";
+		Team team = createTeam(TEAM_ID, "name", "description", null, iconFileHandleId, null, null, null, null);
+        when(mockTeamDAO.get(TEAM_ID)).thenReturn(team);
+		when(mockFileHandleManager.getPreviewFileHandleId(iconFileHandleId)).thenThrow(cause);
+
+		NotFoundException exception = assertThrows(NotFoundException.class, () -> {
+			// Call under test
+			teamManagerImpl.getIconPreviewURL(userInfo, TEAM_ID);
+		});
+
+		assertEquals("No preview was found for the icon of the team with id: " + TEAM_ID, exception.getMessage());
+		assertEquals(cause, exception.getCause());
+		assertEquals(cause.getMessage(), exception.getCause().getMessage());
```

Comment:
Note that this is not necessary as the previous assertEquals will implicitly perform this check.